crates/router/src/connector/nuvei/transformers.rs

```diff
@@ -428,43 +447,74 @@ impl<F>
-                .unwrap_or_default(),
-            time_stamp: time_stamp.clone(),
-            session_token,
-            checksum: encode_payload(vec![
-                merchant_id,
-                merchant_site_id,
-                client_request_id,
-                item.request.amount.to_string(),
-                item.request.currency.to_string(),
-                time_stamp,
-                merchant_secret,
-            ])?,
-            ..request_data
+            connector_auth_type: item.connector_auth_type.clone(),
+            client_request_id: item.attempt_id.clone(),
+            session_token: data.1,
+            capture_method: item.request.capture_method,
+            ..Default::default()
+        })?;
+        Ok(Self {
+            is_rebilling: request_data.is_rebilling,
+            user_token_id: request_data.user_token_id,
+            related_transaction_id: request_data.related_transaction_id,
+            payment_option: request_data.payment_option,
+            ..request
         })
     }
 }
+
 fn get_card_info<F>(
     item: &types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>,
     card_details: &api_models::payments::Card,
 ) -> Result<NuveiPaymentsRequest, error_stack::Report<errors::ConnectorError>> {
     let browser_info = item.request.get_browser_info()?;
-    let related_transaction_id = if item.request.enrolled_for_3ds {
+    let related_transaction_id = if item.is_three_ds() {
         item.request.related_transaction_id.clone()
     } else {
         None
     };
-    let three_d = if item.request.enrolled_for_3ds {
+    let connector_mandate_id = &item.request.connector_mandate_id();
+    if connector_mandate_id.is_some() {
+        return Ok(NuveiPaymentsRequest {
+            related_transaction_id,
+            is_rebilling: Some("1".to_string()),
+            user_token_id: Some(item.request.get_email()?),
+            payment_option: PaymentOption {
+                user_payment_option_id: connector_mandate_id.clone(),
+                ..Default::default()
+            },
+            ..Default::default()
+        });
+    }
+    let (is_rebilling, additional_params, user_token_id) =
+        match item.request.setup_mandate_details.clone() {
+            Some(mandate_data) => {
+                let details = match mandate_data.mandate_type {
+                    api_models::payments::MandateType::SingleUse(details) => details,
+                    api_models::payments::MandateType::MultiUse(details) => {
+                        details.ok_or(errors::ConnectorError::MissingRequiredField {
+                            field_name: "mandate_data.mandate_type.multi_use",
+                        })?
+                    }
+                };
+                let mandate_meta: NuveiMandateMeta =
+                    utils::to_connector_meta_from_secret(Some(details.get_meta_data()?))?;
+                (
+                    Some("0".to_string()),
```

Comment:
why `zero` is hard-coded here? 

Add the comment for hard-coded data, why we are doing that.

Reply:
will add comment